### ISSUE TITLE
Build failed with M1 Mac

### ISSUE THREAD

**Issue by bigianb** [ps2dev/ps2dev#77]
Build fails on an M1 MacBook. Looks like ld in the binutils package doesn't like aarch64. I guess a linux arm64 would have the same issue (raspberry pi 4 for example).

```
bash-3.2$ ./build-all.sh 
Installing latest environment status
HEAD is now at 4459dab Merge pull request #102 from uyjulian/buildsystem_improve
Already on 'master'
Your branch is up to date with 'origin/master'.
Installing latest environment status
HEAD is now at 484a8dd Merge pull request #12 from uyjulian/buildsystem_improve
Already on 'main'
Your branch is up to date with 'origin/main'.
HEAD is now at 9cca5c1 Added abi options to dvp-as for toolchain compatibility
Already on 'dvp-v2.14'
Your branch is up to date with 'origin/dvp-v2.14'.
creating cache ./config.cache
*** This configuration is not supported in the following subdirectories:
     ld
    (Any other directories should still work fine.)
rm: conftest.dSYM: is a directory
../configure: line 2713: test: too many arguments
../configure: line 2785: test: too many arguments
../configure: line 2857: test: too many arguments
../configure: line 3001: test: too many arguments
../configure: line 3073: test: too many arguments
../configure: line 3145: test: too many arguments
../configure: line 3217: test: too many arguments
../configure: line 3289: test: too many arguments
../configure: line 3370: test: too many arguments
../configure: line 3442: test: too many arguments
../configure: line 3514: test: too many arguments
../configure: line 3586: test: too many arguments
../configure: line 3658: test: too many arguments
../configure: line 3730: test: too many arguments
../configure: line 3802: test: too many arguments
updating cache ./config.cache
creating ./config.status
creating Makefile
sed: file conftest.s1 line 26: unterminated `s' command
make: *** No targets.  Stop.
../scripts/001-binutils.sh: Failed.
../scripts/001-dvp.sh: Failed.
../scripts/001-toolchain.sh: Failed.
```

also with noting that gnu sed is being used:

```
ian@Ians-MBP ps2dev % sed --version
sed (GNU sed) 4.9
```

**Comment by pcm720** [comment]
TL;DR: ARM64 target for Darwin (macOS) is not really supported by binutils and GCC used in ps2sdk toolchains.

----
Looks like the main issue preventing binutils-dvp from being built is that configure scripts used in binutils have no idea what `arm64` is and detect host as `-apple-darwin23.1.0`, breaking everything.

You can "fix" this by manually overriding host detection, adding this after line 37 of `ps2toolchain-dvp/scripts/001-binutils.sh`:
```
if [ $OSVER == Darwin ] && [ $(uname -p) == arm ]; then
   TARG_XTRA_OPTS="--host=arm-apple-darwin"
fi
```
ld will still fail to build, but AFAIK ld from GNU binutils has never supported macOS in the first place. That doesn't prevent the binaries from working, anyway.

----
To build GCC for IOP toolchain, I had to add `./contrib/download_prerequisites` to `ps2toolchain-iop/scripts/002-gcc-stage1.sh` before line 58, otherwise it always complains about not having GMP, MPFR and MPC (despite being already installed via Homebrew): 
`configure: error: Building GCC requires GMP 4.2+, MPFR 2.3.1+ and MPC 0.8.0+.`

You'll also have to disable precompiled headers by commenting out a few lines, see this:
https://github.com/riscv-software-src/homebrew-riscv/issues/47#issuecomment-772306695

Don't forget to remove git fetch/reset/checkout from the script (lines 27-30) to keep your changes from being reset every time you launch it.

----
To build binutils for EE toolchain, I had to add `--with-libgmp-prefix=/opt/homebrew/Cellar/gmp/6.3.0` to configure at line 60 of `ps2toolchain-ee/scripts/001-binutils.sh`, otherwise it couldn't find GMP to configure and build gdb. It still complains about other libraries, but at least they are not required. 

Workarounds for building IOP GCC are also required for building stage1 and stage2 of EE GCC.

----

After all of that I was able to build PS2SDK and make a working build of OPL from source.

To honest, I'm not sure why GCC and binutils can't find any of the libraries (which causes half of the problems here). pkg-config is working properly on my system and resolves all names except mpc for some reason.

**Comment by quisys** [comment]
@pcm720 You are truly a wonderful human. I don't know you, but you made my life a little better. Thank you. I hope you understand the extent to which I appreciate what you've done here.

**Comment by fjtrujy** [comment]
Could be nice to update scripts for allowing M1 machines…

**Comment by fjtrujy** [comment]
This was already solved, please re-open if it is not working for you
